### PR TITLE
bpo-36406: Handle namespace packages in doctest

### PR DESCRIPTION
**Pull request by tirkarthi**
Handle namespace packages in doctest where `__file__` of empty package returns None

https://bugs.python.org/issue36406

### CODE REVIEW

**Comment by tirkarthi** [comment]
Note to self and reviewer : I have originally added a fix to handle filename to None and finder is also documented to accept None. But the previous code seems to always default to `module.__name__` where `__file__` attribute was absent for empty packages previously. So I changed this to `getattr(module, '__file__', module.__name__) or module.__name__` to remove check for filename to be None.

File: Lib/doctest.py
Comment (methane) [inline Lib/doctest.py]:
```py
filename = getattr(module, '__file__', None) or module.__name__
```

Comment (tirkarthi) [reply Lib/doctest.py]: Thanks, this makes it simpler and seems to cover all the cases.

* `__file__` can have a value None for empty packages where it falls back to `module.__name__`. 
* `__file__` attribute may be absent where it uses None and then falls back to `module.__name__`
* `__file__` can have a value.

**Comment by tirkarthi** [comment]
Ping @warsaw for review. This was caused due to a23d30f64bd9c5655cfae7f359d4279c47f6cab3 where `__file__` was set as `None` for empty packages causing doctest to fail for empty packages.

File: Lib/doctest.py
Comment (brettcannon) [inline Lib/doctest.py]:
```suggestion
            # __file__ can be None for empty packages.
```

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
@warsaw anything to say before this gets merged?

@tirkarthi if Barry doesn't say anything in a week just ping me and I will merge this.

**Comment by tirkarthi** [comment]
@brettcannon Ping for merge. Thanks.

**Review by methane — APPROVED** [review]

**Comment by brettcannon** [comment]
@tirkarthi thanks for the ping. I will merge this probably this week or next when I have the time.

**Comment by tirkarthi** [comment]
Thanks @brettcannon and @methane for review :)

**Pull request merged**
